Simulator/Edges/AllEdges.h

```diff
@@ -298,8 +298,8 @@ template <class Archive> void AllEdges::load(Archive &archive, std::uint32_t con
 
    // assigns serialized data to objects
    for (int i = 0; i < requiredSize; i++) {
-      W_[i] = WVector[i];
-      sourceVertexIndex_[i] = sourceVertexLayoutIndexVector[i];
-      destVertexIndex_[i] = destVertexLayoutIndexVector[i];
+      W_.push_back(WVector[i]);
+      sourceVertexIndex_.push_back(sourceVertexLayoutIndexVector[i]);
+      destVertexIndex_.push_back(destVertexLayoutIndexVector[i]);
```

Comment:
I think that serialization and deserialization can work directly with the vector data members, no longer needing to marshall/un-marshall to/from raw arrays.

Reply:
Agreed. Will simplify this.